basics/editor-setup.md

```diff
@@ -23,37 +23,47 @@ The choice of an editor is a very personal thing. There are many
 editors, and most programmers have a favorite editor that they use for
 everything and recommend to everyone.
 
-If you aren't sure about which editor you should use, I recommend
-Porcupine. It's a simple editor I wrote in Python; it lets you edit
-files and it doesn't have too many other features. [Install it with these
-instructions](https://github.com/Akuli/porcupine/#installing-porcupine),
-and then [learn to use it by writing the classic Hello World
-program](https://github.com/Akuli/porcupine/wiki/First-Program). Then
-you can [skip the rest of this chapter](#editor-or--prompt).
-
-Note that most other editors come with settings that are not suitable
-for writing Python code. _**TODO:** add a link to the old editor setup
-tutorial here._
-
-Most of these editors lack some important features, they have so many
-features that confuse people or they aren't free. You can use these
-editors if you like them, but **these editors are BAD for getting
-started with programming**:
-
-- PyCharm
-- IDLE
-- Emacs
-- Gedit
-- Nano
-- NetBeans
-- Notepad
-- Pluma
-- Spyder
-- Vim
-- Wingware
-
-This list doesn't contain all bad editors, but these are editors that
-people often try to use. If you know a bad editor and you think I should
+The editors can be broadly divided into three:-
+
+#### The Basic Text Editors
+These editors are usually found in-built in the OS as part of its productive suite. They do not have features like
+code compilation/running, auto-completion, syntax highlighting, etc. that makes one's life easier. They are  usually used for relatively simple
```

Comment:
```suggestion
running code, auto-completion, etc. that make programming easier. They are usually used for relatively simple
```

Deleting "syntax highlighting" because more 3 of the 4 editors mentioned below have it.